### ISSUE TITLE
Not getting expected replication error for .net uwp with SSL enabled with multiple SGs

### ISSUE THREAD

**Issue by hrajput89** [couchbase/couchbase-lite-net#1136]
test_multiple_sgs_with_CBLs test is failing for .net UWP when SSL is enabled. It is failing only for .net UWP and passed on all other .net platforms.

### Library Version

2.5.0-180
-->

### .NET Runtime
UWP 2.5.0-180

### Operating System / Device Details
Windows 10 

### Log Output

[172.23.106.148_cbs.zip](https://github.com/couchbase/couchbase-lite-net/files/3032765/172.23.106.148_cbs.zip)
[python client.log](https://github.com/couchbase/couchbase-lite-net/files/3032766/python.client.log)
[SG 2 logs.zip](https://github.com/couchbase/couchbase-lite-net/files/3032767/SG.2.logs.zip)
[sgcollect_info SG1.zip](https://github.com/couchbase/couchbase-lite-net/files/3032768/sgcollect_info.SG1.zip)

### Expected behavior

We should get "POSIXDomain" error for second replicator

### Actual behavior

We are getting -1 for second replicator

### Steps To Reproduce

pytest -s --timeout 3600 --liteserv-version=2.5.0-180 --liteserv-host=10.100.161.140 --liteserv-port=8080 --xattrs --sg-ssl --sync-gateway-version=2.5.0-239 --mode=cc --server-version=6.0.0-1693 --liteserv-platform=net-uwp --create-db-per-test=cbl-test testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways -k test_multiple_sgs_with_CBLs[listener_tests/multiple_sync_gateways-100]

**Comment by borrrden** [comment]
Please post client logs (I noticed that the test server app is still using the deprecated method of setting text log output.  This needs to be updated to use `Database.Log.File` or it will be a no-op as per the compiler warning)

**Comment by borrrden** [comment]
And to put it more simply than this complicated test -> You are getting the wrong error when you try to pull from a SG that is not running when SSL is enabled on UWP?

**Comment by hrajput89** [comment]
Yes, that's correct

**Comment by hrajput89** [comment]
[log_1554201097.84114.zip](https://github.com/couchbase/couchbase-lite-net/files/3033507/log_1554201097.84114.zip)
CBL Logs

**Comment by borrrden** [comment]
Unable to open the zip file.  It looks corrupt somehow.

**Comment by hrajput89** [comment]
Can you try renaming extension to rar, I changed to zip as GitHub doesn't allow rar to be uploaded as extension

**Comment by borrrden** [comment]
The logs all indicated that POSIX 107 was the reason for the stop.  @Sandychuang8 also tried this and got an error of POSIX 107.  The python logs don't show what was received instead.  What error is being thrown?

**Comment by borrrden** [comment]
I stripped this test down to its essentials and what I found is this:

Since the test always uses continuous replication the test would just loop forever trying to reconnect since continuous replications should not give up.  This was true for both .NET Core and UWP.  I modified the test further to use non-continuous for the final replications and it failed once because it received a "timeout" instead of "connection refused" (The former is a network error and the latter is a posix error but they are both reasonable errors to receive).  On subsequent runs it passes on both .NET Core and UWP multiple times.  I don't think there is an issue here.
